src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -214,3 +230,93 @@ HRESULT OpenTerminalHere::EnumSubCommands(IEnumExplorerCommand** ppEnum)
     *ppEnum = nullptr;
     return E_NOTIMPL;
 }
+
+std::wstring OpenTerminalHere::_GetPathFromExplorer() const
+{
+    using namespace std;
+    using namespace winrt;
+
+    wstring path;
+    HRESULT hr = NOERROR;
+
+    auto hwnd = ::GetForegroundWindow();
+    if (hwnd == nullptr)
+    {
+        return path;
+    }
+
+    TCHAR szName[MAX_PATH] = { 0 };
+    ::GetClassName(hwnd, szName, MAX_PATH);
+    if (0 == StrCmp(szName, L"WorkerW") ||
+        0 == StrCmp(szName, L"Progman"))
+    {
+        //special folder: desktop
+        hr = ::SHGetFolderPath(NULL, CSIDL_DESKTOP, NULL, SHGFP_TYPE_CURRENT, szName);
+        if (FAILED(hr))
+        {
+            return path;
+        }
+
+        path = szName;
+        return path;
+    }
+
+    if (0 != StrCmp(szName, L"CabinetWClass"))
+    {
+        return path;
+    }
+
+    auto shell = create_instance<IShellWindows>(CLSID_ShellWindows);
+    if (shell == nullptr)
+    {
+        return path;
+    }
+
+    com_ptr<IDispatch> disp;
+    wil::unique_variant variant;
+    variant.vt = VT_I4;
+
+    com_ptr<IWebBrowserApp> browser;
+    // look for correct explorer window
+    for (variant.intVal = 0;
+         shell->Item(variant, disp.put()) == S_OK;
+         variant.intVal++)
+    {
+        com_ptr<IWebBrowserApp> tmp;
+        if (FAILED(disp->QueryInterface(browser.put())))
+        {
+            disp = nullptr;

```

Comment:
Do we really need to be manually resetting `disp` to `nullptr` each time? Won't it get cleaned up correctly just by virtue of being a `com_ptr`?

Reply:
when get_HWND is failed hwnd is nullptr, it will fail the next == comparison,  it will not affect the algorithm

Reply:
you are right this is bug. 
disp no need to set to nullptr, i traced the source code, the existing pointer got released in operator =